packages/protobuf/src/equals.ts

```diff
@@ -64,33 +118,35 @@ function fieldEquals(
     case "enum":
       return a.get(f) === b.get(f);
     case "message":
-      return reflectEquals(a.get(f), b.get(f));
+      return reflectEquals(a.get(f), b.get(f), opts);
     case "map": {
-      const ma = a.get(f);
-      const mb = b.get(f);
-      const keysA: unknown[] = [];
-      for (const k of ma.keys()) {
-        if (!mb.has(k)) {
+      const mapA = a.get(f);
+      const mapB = b.get(f);
+      const keys: unknown[] = [];
+      for (const k of mapA.keys()) {
+        if (!mapB.has(k)) {
           return false;
         }
-        keysA.push(k);
+        keys.push(k);
       }
-      for (const k of mb.keys()) {
-        if (!ma.has(k)) {
+      for (const k of mapB.keys()) {
+        if (!mapA.has(k)) {
           return false;
         }
       }
```

Comment:
You have a more efficient/concise way to test this below, for extensions: instead of scanning through both sets of keys, check that they both have the same _number_ of keys and then just scan `a`, verifying that each key is also present in `b`.

Reply:
This is a facade to access an object like a map type. I think the logic is trying to avoid some expensive operations. I've added a comment.